docs/bokeh/source/docs/first_steps/first_steps_9.rst

```diff
@@ -34,44 +34,44 @@ Follow these steps to recreate the example from above:
 1. Import the ``Div``, ``RangeSlider``, and ``Spinner`` widgets
```

Comment:
Similar to my other comment, we need to use multiples of 3 (not 4) spaces for indentations. This section would then look like this:

```
1. Import the ``Div``, ``RangeSlider``, and ``Spinner`` widgets
   from ``bokeh.models``:

   .. code-block:: python

      from bokeh.models import Div, RangeSlider, Spinner

2. Set up your data, your figure, and your renderer:

   .. code-block:: python

      x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
      y = [4, 5, 5, 7, 2, 6, 4, 9, 1, 3]

      p = figure(x_range=(1,9), width=500, height=250)
      points = p.circle(x=x, y=y, size=30, fill_color="#21a7df")

3. Create a ``Div`` object and pass it some HTML code as a string:

   .. code-block:: python

      div = Div(
          text="""
              <p>Select the circle's size using this control element:</p>
              """,
          width=200,
          height=30,
      )

4. Create a ``Spinner`` object:

   .. code-block:: python

      spinner = Spinner(
          title="Circle size",  # a string to display above the widget
          low=0,  # the lowest possible number to pick
          high=60,  # the highest possible number to pick
          step=5,  # the increments by which the number can be adjusted
          value=points.glyph.size,  # the initial value to display in the widget
          width=200,  #  the width of the widget in pixels
          )

5. To link the value generated by the spinner to the ``size`` property of your
   glyph, use the :func:`~bokeh.model.Model.js_link` function. This function
   uses JavaScript to interactively link two Bokeh models. The first argument
   you pass to this function is the attribute of your spinner (``"value"``)
   that you want to link to your circle glyphs. The second attribute is the
   glyph you want to link to your spinner (``points.glyph``). The third
   argument is the property of your glyph that you want to link to your
   spinner's ``value``:

   .. code-block:: python

      spinner.js_link("value", points.glyph, "size")

6. Create a RangeSlider object:

   .. code-block:: python

      range_slider = RangeSlider(
          title="Adjust x-axis range", # a title to display above the slider
          start=0,  # set the minimum value for the slider
          end=10,  # set the maximum value for the slider
          step=1,  # increments for the slider
          value=(p.x_range.start, p.x_range.end),  # initial values for slider
          )

7. To link the values generated by the RangeSlider to the existing plot, use the
   :func:`~bokeh.model.Model.js_link` function again. This time, however, you
   need to assign two values at once: the beginning of the plot's x-axis and
   the end of its x-axis. The RangeSlider returns a tuple of those two values.
   Therefore, you need to use the ``attr_selector`` of the ``js_link()``
   function to tell Bokeh which part of the Tuple to assign to either the start
   or the end of the plot's x-axis:

   .. code-block:: python

      range_slider.js_link("value", p.x_range, "start", attr_selector=0)
      range_slider.js_link("value", p.x_range, "end", attr_selector=1)

8. Create a layout with all the elements of your dashboard and display it in a
   browser:

   .. code-block:: python

      layout = layout([
          [div, spinner],
          [range_slider],
          [p],
      ])

      show(layout)

```

Reply:
@tcmetzger Thanks for your hint, you saved me some time. I was going to spent some more time on this. I will change this in today.